### PR TITLE
Update toil to 5.12.0

### PR DESCRIPTION
**Pull request by mencian**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by mencian** [comment]
@BiocondaBot please add label

**Review by hechth — COMMENTED** [review]
Thanks for updating this, please see the inline comments.

File: recipes/toil/meta.yaml [hunk 1]
```diff
@@ -56,24 +61,25 @@ requirements:
...
+    - pynacl >=1.4.0
     #google reqs
-    - apache-libcloud >=2.2.1,<3
+    - apache-libcloud >=2.2.1
     - google-cloud-storage >=2,<=2.8.0
     #cwl reqs
-    - cwltool ==3.1.20230425144158
-    - schema-salad >=8.4.20230128170514,<9
+    - cwltool ==3.1.20230601100705
+    - schema-salad >=8.4.20230128170514
```
[the remark this answers is not in the document]
Comment (mencian) [reply recipes/toil/meta.yaml:70]: It seems these very strict pins are necessary; I've edit the version constraints based on [DataBiosphere/toil/requirements-cwl.txt](https://github.com/DataBiosphere/toil/blob/master/requirements-cwl.txt).

File: recipes/toil/meta.yaml [hunk 2]
```diff
@@ -56,24 +61,25 @@ requirements:
...
     - galaxy-tool-util
     - ruamel.yaml >=0.15,<=0.17.21
```
[the remark this answers is not in the document]
Comment (mencian) [reply recipes/toil/meta.yaml:72]: The version restraints for 'ruamel.yaml' are from the upstream project's requirements-cwl.txt. Please see my previous comment.

**Review by hechth — APPROVED** [review]

**Pull request merged**